README.md

```diff
@@ -102,55 +98,77 @@ So, considering you did `git add file1.ext file2.ext`, lint-staged will run the
+When supplying configuration in JS format it is possible to define the linter command as a function which receives an array of staged filenames/paths and returns the complete linter command as a string. It is also possible to return an array of complete command strings, for example when the linter command supports only a single file input.
 
-#### `package.json` example with `ignore` option:
+```ts
+type LinterFn = (filenames: string[]) => string | string[]
+```
 
-```json
-{
-  "lint-staged": {
-    "linters": {
-      "*.{js,scss}": ["some command", "git add"]
-    },
-    "ignore": ["**/dist/*.min.js"]
-  }
+### Example: Wrap filenames in single quotes and run once per file
+
+```js
+// .lintstagedrc.js
+module.exports = {
+  '**/*.js?(x)': filenames => filenames.map(filename => `prettier --write '${filename}'`)
 }
 ```
 
-Notice that the linting commands now are nested into the `linters` object. The following options are available for advanced configuration:
+### Example: Run `tsc` on changes to TypeScript files, but do not pass any filename arguments
 
-#### Options
+```js
+// lintstaged.config.js
+module.exports = {
+  '**/*.ts?(x)': () => 'tsc -p tsconfig.json --noEmit'
+}
+```
 
-* `concurrent` — _true_ — runs linters for each glob pattern simultaneously. If you don’t want this, you can set `concurrent: false`
-* `chunkSize` — Max allowed chunk size based on number of files for glob pattern. This option is only applicable on Windows based systems to avoid command length limitations. See [#147](https://github.com/okonet/lint-staged/issues/147)
-* `globOptions` — `{ matchBase: true, dot: true }` — [micromatch options](https://github.com/micromatch/micromatch#options) to
-  customize how glob patterns match files.
-* `ignore` - `['**/docs/**/*.js']` - array of glob patterns to entirely ignore from any task.
-* `linters` — `Object` — keys (`String`) are glob patterns, values (`Array<String> | String`) are commands to execute.
-* `subTaskConcurrency` — `1` — Controls concurrency for processing chunks generated for each linter. This option is only applicable on Windows. Execution is **not** concurrent by default(see [#225](https://github.com/okonet/lint-staged/issues/225))
-* `relative` — `false` — if `true` it will give the relative path from your `package.json` directory to your linter arguments.
+### Example: Run eslint on entire repo if more than 10 staged files
 
-## Filtering files
+```js
+// .lintstagedrc.js
+module.exports = {
+  '**/*.js?(x)': filenames => (filenames.length > 10 ? 'eslint .' : `eslint ${filenames.join(' ')}`)
+}
+```
+
+### Example: Use your own globs
 
-It is possible to run linters for certain paths only by using glob patterns. [micromatch](https://github.com/micromatch/micromatch) is used to filter the staged files according to these patterns. File patterns should be specified _relative to the `package.json` location_ (i.e. where `lint-staged` is installed).
+```js
+// lintstaged.config.js
```

Comment:
Should be `lint-staged.config.js`

Reply:
Thanks, seems like when it's a dotfile, the dash can be omitted.